apps/OpenSpace/ext/launcher/src/sgctedit/sgctedit.cpp

```diff
@@ -0,0 +1,139 @@
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "sgctedit/sgctedit.h"
+
+SgctEdit::SgctEdit(QWidget* parent, std::vector<sgct::config::Window>& windowList,
+                   sgct::config::Cluster& cluster, const QList<QScreen*> screenList)
+    : QDialog(parent)
+    , _cluster(cluster)
+    , _windowList(windowList)
+{
+    systemMonitorConfiguration(screenList);
+    setWindowTitle("Display/Window Configuration Editor");
+    createWidgets();
+}
+
+void SgctEdit::systemMonitorConfiguration(const QList<QScreen*> screenList) {
+    size_t nScreensManaged = std::min(screenList.length(), 2);
+    for (size_t s = 0; s < nScreensManaged; ++s) {
+        int actualWidth = std::max(screenList[s]->size().width(),
+            screenList[s]->availableGeometry().width());
+        int actualHeight = std::max(screenList[s]->size().height(),
+            screenList[s]->availableGeometry().height());
+        _monitorSizeList.push_back({
+            screenList[s]->availableGeometry().x(),
+            screenList[s]->availableGeometry().y(),
+            actualWidth,
+            actualHeight
+        });
+    }
+    _nMaxWindows = (_monitorSizeList.size() == 1) ? 3 : 4;
+}
+
+void SgctEdit::createWidgets() {
+    QVBoxLayout* layoutMainV = new QVBoxLayout(this);
+    QHBoxLayout* layoutMainH = new QHBoxLayout;
+    _orientationWidget = new Orientation();
+
+    {
+        _monBox = new MonitorBox(
+            _monitorWidgetSize,
+            _monitorSizeList,
+            _nMaxWindows,
+            _colorsForWindows
+        );
+        QHBoxLayout* layoutMonBox = new QHBoxLayout();
+        layoutMonBox->addStretch(1);
+        layoutMonBox->addWidget(_monBox);
+        layoutMonBox->addStretch(1);
+        layoutMainV->addLayout(layoutMonBox);
+        _monBox->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+        _monBox->setFixedSize(_monitorWidgetSize.width(), _monitorWidgetSize.height());
+        addDisplayLayout(_monBox, layoutMainH);
+    }
+    {
+        layoutMainV->addLayout(layoutMainH);
+        _orientationWidget->addButtonToLayout(layoutMainV);
+        _fileSupportWidget = new FileSupport(
+            layoutMainV,
+            _monitorSizeList,
+            _displayWidget,
+            _orientationWidget,
+            _windowList,
+            _cluster,
+            [this](bool accepted) {
+                if (accepted) {
+                    _saveSelected = true;
+                    accept();
+                }
+                else {
+                    reject();
+                }
+            }
+        );
+    }
+}
+
+void SgctEdit::addDisplayLayout(MonitorBox* monBox, QHBoxLayout* layout)
```

Comment:
This function knows about the `_monBox`, so it can just use that member variable directly, rather than needing it to be passed as a function parameter